pep-0718.rst

```diff
@@ -0,0 +1,152 @@
+PEP: 718
+Title: Subscriptable functions
+Author: James Hilton-Balfe <gobot1234yt@gmail.com>
+Sponsor: Guido van Rossum <guido@python.org>
+Discussions-To: https://discuss.python.org/t/26463/
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Content-Type: text/x-rst
+Created: 23-Jun-2023
+Python-Version: 3.13
+Post-History: `24-Jun-2023 <https://discuss.python.org/t/28457/>`__
+
+Abstract
+--------
+
+This PEP proposes making ``FunctionType``\ , ``MethodType``\ , ``BuiltinFunctionType``
+and ``BuiltinMethodType`` instances subscriptable for typing purposes. Doing so allows
+for specialisation in complex, hard-to-infer situations for type checkers where
+bi-directional inference (which allows for the types of parameters of anonymous
+functions to be inferred) is insufficient.
+
+Motivation
+----------
+
+Currently, it is not possible to infer certain situations for generic functions where
+unspecialised literals are involved
```

Comment:
This shouldn't say unspecialised literals, I think this is a remnant of the example foo being at the top of this section